### PR TITLE
Feat/open telemetry

### PR DESCRIPTION
**Pull request by lewinskimaciej**
Adds OpenTelemetry sending traces, metrics and logs to Grafana Cloud

### CODE REVIEW

File: .github/workflows/release-to-preview.yml [hunk 1]
```diff
@@ -121,7 +121,7 @@ jobs:
       - name: Check service status
         id: get-service-status
         run: |
-          response=$(ssh -o StrictHostKeyChecking=no -J ubuntu@${{ secrets.SSH_JUMPHOST }} ubuntu@${{ secrets[matrix.node] }} "sleep 10; echo \$(curl -o /dev/null -s -w %{http_code} localhost:3000)")
+          response=$(ssh -o StrictHostKeyChecking=no -J ubuntu@${{ secrets.SSH_JUMPHOST }} ubuntu@${{ secrets[matrix.node] }} "sleep 15; echo \$(curl -o /dev/null -s -w %{http_code} localhost:3000)")
```
Comment (RobbyUitbeijerse) [inline .github/workflows/release-to-preview.yml:124]: funny, this actually helps?

File: packages/indexer/package.json [hunk 2]
```diff
@@ -40,7 +56,7 @@
     "@types/hapi__hapi": "^20.0.9",
     "@types/hapi__inert": "^5.2.3",
     "@types/hapi__vision": "^5.5.3",
-    "@types/ioredis": "^4.28.1",
+    "@types/ioredis": "4.28.5",
```
Comment (RobbyUitbeijerse) [inline packages/indexer/package.json:59]: do you still need this for something specific?

File: packages/indexer/src/orderbook/orders/seaport-v1.1/index.ts [hunk 3]
```diff
@@ -705,15 +705,17 @@ export const save = async (
...
       limit(async () =>
-        tracer.trace("handleOrder", { resource: "seaportSave" }, () =>
-          handleOrder(
-            orderInfo.orderParams as Sdk.SeaportBase.Types.OrderComponents,
-            orderInfo.metadata,
-            orderInfo.isReservoir,
-            orderInfo.isOpenSea,
-            orderInfo.openSeaOrderParams
+        trace
+          .getTracer("seaport-v1.1")
```
Comment (RobbyUitbeijerse) [inline packages/indexer/src/orderbook/orders/seaport-v1.1/index.ts:709]: is the 1.1 correct here?

File: packages/indexer/src/orderbook/orders/seaport-v1.5/index.ts [hunk 4]
```diff
@@ -926,17 +926,19 @@ export const save = async (
...
-            orderInfo.orderParams as Sdk.SeaportBase.Types.OrderComponents,
-            orderInfo.metadata,
-            orderInfo.isReservoir,
-            orderInfo.isOpenSea,
-            orderInfo.isOkx,
-            orderInfo.isImtbl,
-            orderInfo.openSeaOrderParams
+        trace
+          .getTracer("seaport-v1.4")
+          .startActiveSpan("seaportV15Save", () =>
```
Comment (RobbyUitbeijerse) [inline packages/indexer/src/orderbook/orders/seaport-v1.5/index.ts:931]: Shouldn't this be 1.5? `seaportV15Save` seems to suggest Seaport 1.5, not 1.4

File: .github/workflows/release-to-preview.yml (hunk 1 above)
Comment (lewinskimaciej) [reply .github/workflows/release-to-preview.yml:124]: It's actually Vlad's change, idk why it's showing here cause I rebased onto develop ¯\_(ツ)_/¯

File: packages/indexer/package.json (hunk 2 above)
Comment (lewinskimaciej) [reply packages/indexer/package.json:59]: i've just updated ioredis as a way of testing, i could roll back but this doesn't have any breaking changes so I just decided to keep it updated. Version 5+ has a lot of breaking changes(quite easy to fix as far as I saw) but I didn't want to do that here

File: packages/indexer/src/orderbook/orders/seaport-v1.1/index.ts (hunk 3 above)
Comment (lewinskimaciej) [reply packages/indexer/src/orderbook/orders/seaport-v1.1/index.ts:709]: It's just name of the handler, that tracer name doesn't matter that much, i had to pick something so I just used seaport version, each of the handlers has a different one

File: packages/indexer/src/orderbook/orders/seaport-v1.5/index.ts (hunk 4 above)
Comment (lewinskimaciej) [reply packages/indexer/src/orderbook/orders/seaport-v1.5/index.ts:931]: Yeah this should, fixed

File: packages/indexer/src/orderbook/orders/seaport-v1.1/index.ts (hunk 3 above)
Comment (royscheepens) [reply packages/indexer/src/orderbook/orders/seaport-v1.1/index.ts:709]: For future reference; we only use Reservoir's `seaport-1.5` implementation for Beam and Immutable. So don't spend too much time on other versions.

Comment (lewinskimaciej) [reply packages/indexer/src/orderbook/orders/seaport-v1.1/index.ts:709]: I don't, i just had to update usages from existing codebase as DataDog sdk is a bit different

**Comment by royscheepens** [comment]
Oh whoops this should have been merged into develop

**Pull request merged**